### DISCUSSION TITLE
pybricks blocks - try/except

### DISCUSSION THREAD

**Discussion by amberneb** [pybricks/support#1947] [General]
Our team switched to pybricks this year after learning about it at FLL worlds. We are using the blocks version and we are wondering if there is a way to do some of the things that you can do with text-based coding (specifically try: except:) 

The kids searched through all the block options and didn't see a block that they thought would work.

**Comment by laurensvalk** [comment]
To do more advanced things like that, you could consider combining blocks with Python.

You can write any Python function, and import and call it in your block program using the blocks on the yellow tab.

What kind of exception did you want to catch?

**Comment by amberneb** [comment]
Okay, we see the external tasks block. Is that what you are talking about? If so, would you be able to explain in more detail (but in simple terms for this non-text-based-coding-experienced coach) how to do this? I'm feeling very lost!

The problem the kids are having is they are trying to use the motor to lift their attachment. Sometimes it lifts the perfect amount and sometimes it keeps trying to lift even when it can't lift any more and then the program just stops running. 

Thank you!

**Reply by BertLindeman** [reply]
Maybe you do not need a try / exception for this.

In the blocks you can also know if the motor stalls.
E.g. like this:
![image](https://github.com/user-attachments/assets/4c871708-18fc-4a89-af5d-6f4feba70b23)

Not meant as a part of your program, just to show the blocks.

Bert

**Reply by laurensvalk** [reply]
Thanks for the additional context @amberneb! This sounds quite similar to the following question and answer, so I think that would help: https://github.com/orgs/pybricks/discussions/1621
